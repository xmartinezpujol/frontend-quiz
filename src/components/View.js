import styled from 'styled-components';
import PropTypes from 'prop-types';

import COLOR_PALETTE from '../Constants';

const Container = {
  styles: {
    margin: '0 14px',
    maxWidth: '100%',
    '@media(min-width: 768px)': {
      margin: '0 auto',
      width: 720,
    },
    '@media(min-width: 992px)': {
      margin: '0 auto',
      width: 920,
    },
    '@media(min-width: 1200px)': {
      margin: '0 auto',
      width: '90vw',
      maxWidth: 1430,
    },
  },
  default: {},
};

const View = styled.div(props => (
  {
    borderRadius: props.round ? props.round : 0,
    display: 'flex',
    width: props.width ? props.width : 'auto',
    height: props.height ? props.height : 'auto',
    background: COLOR_PALETTE[props.type],
    justifyContent: props.justify,
    alignItems: props.align,
    flexDirection: props.direction,
    ...Container[props.container ? 'styles' : 'default'],
    padding: props.padding,
  }
));

View.propTypes = {
  type: PropTypes.string,
  round: PropTypes.number,
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  justify: PropTypes.string,
  align: PropTypes.string,
  direction: PropTypes.string,
  container: PropTypes.bool,
  padding: PropTypes.number,
};

export default View;
