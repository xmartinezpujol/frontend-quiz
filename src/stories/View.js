import React from 'react';

import { storiesOf } from '@storybook/react';
import {
  withKnobs,
  text,
  number,
  select,
} from '@storybook/addon-knobs';

import View from '../components/View';

const optionsView = {
  transparent: 'transparent',

  // Primary
  red: 'red',
  red_light: 'red_light',
  blue: 'blue',
  blue_light: 'blue_light',
  background_text: 'background_text',
  text: 'text',
  text2: 'text2',
  title: 'title',
};

storiesOf('View', module).addDecorator(withKnobs)
  .add('colors', (() => {
    const label = 'Color';
    const defaultValue = 'consumer';
    const value = select(label, optionsView, defaultValue);
    return (
      <View type="purewhite" container direction="column" >
        <div style={{ marginLeft: 20, marginBottom: 50 }}>
          <h1>Test View</h1>
          <View
            direction="column"
            justify="center"
            align="center"
            round={number('Round', 20)}
            width={text('Width', '100%')}
            height={text('Height', '200')}
            type={value}
            padding={number('Padding', 20)}
          >
            <p>{text('Message', 'Text here...')}</p>
          </View>
        </div>
        <div style={{ marginLeft: 20 }}>
          <h1>Secondary Colours</h1>
        </div>
        <div
          style={{
            display: 'flex',
            flexFlow: 'wrap row',
            justifyContent: 'center',
            marginBottom: 50,
          }}
        >
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View direction="column" justify="center" align="center" type="red" height="200" width="200">
              <p>#F83E59</p>
              <p>RED</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View direction="column" justify="center" align="center" type="red_light" height="200" width="200">
              <p>#FD5E74</p>
              <p>RED LIGHT</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View direction="column" justify="center" align="center" type="blue" height="200" width="200">
              <p>#11CFDA</p>
              <p>BLUE</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View direction="column" justify="center" align="center" type="blue_light" height="200" width="200">
              <p>#71D6E2</p>
              <p>BLUE LIGHT</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View direction="column" justify="center" align="center" type="title" height="200" width="200">
              <p>#36424A</p>
              <p>TITLE</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View direction="column" justify="center" align="center" type="text" height="200" width="200">
              <p>#6E7A83</p>
              <p>TEXT</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View direction="column" justify="center" align="center" type="text2" height="200" width="200">
              <p>#B6BABD</p>
              <p>TEXT2</p>
            </View>
          </div>
        </div>
        <div style={{ marginLeft: 20 }}>
          <h1>Infrastructure Colors</h1>
        </div>
        <div
          style={{
            display: 'flex',
            flexFlow: 'wrap row',
            justifyContent: 'center',
            marginBottom: 50,
          }}
        >
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View justify="center" align="center" type="purewhite" height="200" width="200">
              <p>#FFFFFF</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View justify="center" align="center" type="spacewhite" height="200" width="200">
              <p>#FAFAFA</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View justify="center" align="center" type="night" height="200" width="200">
              <p>#1A1D1F</p>
            </View>
          </div>
        </div>
        <div
          style={{
            display: 'flex',
            flexFlow: 'wrap row',
            justifyContent: 'center',
            marginBottom: 50,
          }}
        >
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View justify="center" align="center" type="positive" height="200" width="200">
              <p>#86D9C5</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View justify="center" align="center" type="negative" height="200" width="200">
              <p>#EF8B8B</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View justify="center" align="center" type="temporary" height="200" width="200">
              <p>#FBB86D</p>
            </View>
          </div>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <View justify="center" align="center" type="neutral" height="200" width="200">
              <p>#6E7A83</p>
            </View>
          </div>
        </div>
      </View>
    );
  }));
