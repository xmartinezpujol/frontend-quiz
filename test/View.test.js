import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import View from '../src/components/View';

configure({ adapter: new Adapter() });

describe('View', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <View>
        {"I'm just some text inside a viewport"}
      </View>,
    );
  });
  it('View renders correctly', () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
});
