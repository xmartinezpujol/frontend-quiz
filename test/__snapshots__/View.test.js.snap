// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View View renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1ysjyam",
    },
    "children": Array [
      Object {
        "data": "I'm just some text inside a viewport",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
